test/sideinput-source-e2e/testdata/sideinput_source.yaml

```diff
@@ -0,0 +1,46 @@
+apiVersion: numaflow.numaproj.io/v1alpha1
+kind: Pipeline
+metadata:
+  name: sideinput-source
+spec:
+  sideInputs:
+    - name: myticker
+      container:
+        image: "quay.io/shubhamdixit863/sideinput-sink-e2e-test:latest"
+        imagePullPolicy: Always
+      trigger:
+        schedule: "* * * * *"
+        # timezone: America/Los_Angeles
+  vertices:
+    - name: in
+      scale:
+        min: 1
+      source:
+        udsource:
+          container:
+            image: quay.io/shubhamdixit863/simple_source_with_sideinput:latest
+      sideInputs:
+        - myticker
+    - name: cat
+      udf:
+        builtin:
+          name: cat # A built-in UDF which simply cats the message      -
+    - name: log-sink
+      sink:
+        log: {}
+    - name: redis-uds
+      sink:
+        udsink:
+          container:
+            image: quay.io/numaio/numaflow-sink/redis-e2e-test-sink:v0.5.0
+      sideInputs:
+        - myticker
```

Comment:
Can we remove `cat` and `log-sink` vertices? Since we are testing source with sideinput, it's better to keep other parts of the pipeline as simple as possible.